yt_dlp/extractor/ciscolive.py

```diff
@@ -2,144 +2,179 @@
+            'creators': 'count:4',
+            'description': 'md5:70ee6edf45c8221d7e9346426ae25fb0',
+            'display_id': '1749603837855001eWhB',
+            'duration': 2921.387,
+            'series': 'Cisco Live On Demand',
+            'series_id': '1749591952508001pGUf',
+            'tags': [],
+            'thumbnail': r're:https?://cf-images\.us-east-1\.prod\.boltdns\.net/.+\.jpg',
+            'timestamp': 1749602136,
+            'upload_date': '20250611',
+            'uploader_id': '5647924234001',
+        },
     }, {
-        'url': 'https://www.ciscolive.com/global/on-demand-library.html?#/session/1490051371645001kNaS',
-        'only_matching': True,
+        'url': 'https://www.ciscolive.com/on-demand/on-demand-library.html?search.sessiontype=Keynote#/session/1748956425383001aEIK',
+        'info_dict': {
+            'id': '6373626841112',
+            'ext': 'mp4',
+            'title': 'Keynote: Innovation in Action - KEYGEN-1002',
+            'creators': 'count:6',
+            'description': 'md5:3a1d82ca9e4e3505e251c251d3509cc5',
+            'display_id': '1748956425383001aEIK',
+            'duration': 4405.739,
+            'series': 'Cisco Live On Demand',
+            'series_id': '1748956425383001aEIK',
+            'tags': [],
+            'thumbnail': r're:https?://cf-images\.us-east-1\.prod\.boltdns\.net/.+\.jpg',
+            'timestamp': 1748553118,
+            'upload_date': '20250529',
+            'uploader_id': '5647924234001',
+        },
     }]
 
     def _real_extract(self, url):
-        rf_id = self._match_id(url)
-        rf_result = self._call_api('session', rf_id, {'id': rf_id}, url)
-        return self._parse_rf_item(rf_result['items'][0])
+        url_type, display_id = self._match_valid_url(url).group('type', 'id')
+        if url_type == 'video':
+            file = self._call_api('file', display_id)
+            rf_id = traverse_obj(file, ('items', 'file', 'rainfocusId', {str}))
+        else:
+            rf_id = display_id
+
+        return {
+            '_type': 'url_transparent',
+            'ie_key': 'BrightcoveNew',
+            'display_id': display_id,
+            **traverse_obj(self._call_api('session', rf_id), ('items', ..., {
+                'title': ('title', {clean_html}),
+                'creators': ('participants', ..., 'fullName', {str}),
+                'description': ('abstract', {clean_html}),
+                'series': ('eventName', {clean_html}),
+                'series_id': ('sessionID', {str}),
+                'url': ('videos', ..., 'url', {str_or_none}, {lambda x: self.BRIGHTCOVE_URL_TEMPLATE % x}, any),
+            }, any)),
+        }
 
 
 class CiscoLiveSearchIE(CiscoLiveBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?ciscolive(?:\.cisco)?\.com/(?:global/)?on-demand-library(?:\.html|/)'
+    IE_NAME = 'ciscolive:search'
+
+    _VALID_URL = r'https?://(?:www\.)?ciscolive\.com/on-demand/on-demand-library\.html'
```

Comment:
This could be disambiguated and then scrap the `suitable()` method, perhaps:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?ciscolive\.com/on-demand/on-demand-library\.html\?\w(?:(?!#/session/).)+'
```

Reply:
https://www.ciscolive.com/on-demand/on-demand-library.html?search.sessiontype=Keynote#/session/1748956425383001aEIK

Reply:
That matches the session IE, doesn't it? A URL with a `#/session/` query parameter (or now, `#/video/`) should find the session. Maybe you can formulate a query with a `#/session/`  parameter and other parameters that produces no result on the website (wrong `sessionType`, eg) but gets one with the extractor: that reproduces the existing behaviour (when it used to work).

Otherwise just have the search IE and an internal IE that accepts, say, `ciscolive:video:...` and `ciscolive:session:...` pseudo-URLs.